scene/animation/animation_node_state_machine.cpp

```diff
@@ -223,8 +224,16 @@ void AnimationNodeStateMachinePlayback::_set_current(AnimationNodeStateMachine *
 		group_end_transition = Ref<AnimationNodeStateMachineTransition>();
 	}
 
-	// Validation.
+	// Validate grouped child state machine and ensure signals are connected for propagation.
 	if (anodesm->get_state_machine_type() == AnimationNodeStateMachine::STATE_MACHINE_TYPE_GROUPED) {
+		Ref<AnimationNodeStateMachinePlayback> playback = p_state_machine->get_animation_tree()->get(base_path + current + "/playback");
+		Callable callable = callable_mp(this, &AnimationNodeStateMachinePlayback::_propagate_grouped_child_signal);
+		if (!playback->is_connected(SceneStringName(state_started), callable)) {
+			String concatenated = vformat("%s/", current);
+			playback->connect(SceneStringName(state_started), callable.bind(concatenated, true));
+			playback->connect(SceneStringName(state_finished), callable.bind(concatenated, false));
+		}
+
```

Comment:
It is a not good way to connect signals every time you change the current. Also, it looks like the connection will remain even after changing to a state machine type other than Grouped.

There are two possible ways to do this correctly.

1. GroupedStateMachine directly calls the parent's Playback function without using a signal, as in https://github.com/godotengine/godot/pull/88179/files#diff-9176faca3ecbebad6f340de9260278b31819bdfae318c1cd310a01c60229d0e7R202-R210.

2. Connect the signal. However, to avoid doing this too often, you should do it in `set_state_machine_type()` and `_tree_changed()` and disconnect the signals when it is not needed.

Reply:
I ended up adding this logic to `_set_grouped` so it will connect/disconnect whenever the playback instance mode is changed. At first I preferred option 1 but the logic for `_get_parent_playback` seemed much more expensive vs `is_connected`. For option 2 I wasn't sure how to update all playback instances from the non-playback node outside of a process context, and even then I _think_ it would require a check in process anyways so checking if the value changes in `_set_grouped` seemed like the efficient approach. I still need to test that it handles if the state machine type changes at runtime but it handles the basic case at least.